### PR TITLE
[test/e2e] Enhancement TopoMatcher

### PR DESCRIPTION
**Pull request by arugal**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [ ] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues

___
Calls expected-data to support variable.

### CODE REVIEW

File: test/e2e/e2e-base/src/main/java/org/apache/skywalking/e2e/assertor/VariableExpressParser.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+/**
+ * @author zhangwei
+ */
+public class VariableExpressParser {
+
+
+    public static <T> T parse(String express, List<T> actual, Function<T, String> getFiled) {
+        express = express.trim();
+        if (!express.startsWith("${") && !express.endsWith("}")) {
```
Comment (kezhenxu94) [inline test/e2e/e2e-base/src/main/java/org/apache/skywalking/e2e/assertor/VariableExpressParser.java]:
```suggestion
        if (!(express.startsWith("${") && express.endsWith("}"))) {
```

**Comment by arugal** [comment]
@wu-sheng  Hi, look at these two.
https://github.com/apache/skywalking/blob/3cb622daed914761cbb791d34d78ea7d9b8e5d5f/test/e2e/e2e-base/src/main/java/org/apache/skywalking/e2e/topo/CallMatcher.java#L38-L43

```java
    private static void convertNodeId(List<CallMatcher> callMatchers, List<Node> nodes) {
        for (CallMatcher callMatcher : callMatchers) {
            Node sourceNode = VariableExpressParser.parse(callMatcher.getSource(), nodes, Node::getName);
            Node targetNode = VariableExpressParser.parse(callMatcher.getTarget(), nodes, Node::getName);

            boolean convert = false;
            if (nonNull(sourceNode)) {
                callMatcher.setSource(sourceNode.getId());
                convert = true;
            }
            if (nonNull(targetNode)) {
                callMatcher.setTarget(targetNode.getId());
                convert = true;
            }

            if (convert) {
                callMatcher.setId(String.join("_", callMatcher.getSource(), callMatcher.getTarget()));
            }
        }
    }
```

**Comment by wu-sheng** [comment]
Got it. And confirmed.

**Comment by wu-sheng** [comment]
@arugal e2e just fails, I triggered re-run.

**Pull request merged**